sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1058,7 +1064,7 @@ public TextAnalyticsClient(Uri endpoint, AzureKeyCredential credential, TextAnal
         /// <exception cref="RequestFailedException">Service returned a non-success
         /// status code.</exception>
         public virtual async Task<Response<AnalyzeSentimentResultCollection>> AnalyzeSentimentBatchAsync(IEnumerable<TextDocumentInput> documents, AnalyzeSentimentOptions options = default, CancellationToken cancellationToken = default) =>
-            await _serviceClient.AnalyzeSentimentBatchAsync(documents, options, cancellationToken).ConfigureAwait(false);
+            await _serviceClient.AnalyzeSentimentBatchAsync(documents, options?.CheckSupported(ServiceVersion), cancellationToken).ConfigureAwait(false);
```

Comment:
What if we create an internal shared source `IVersioningValidation` interface and then file a bug to create an analyzer making sure it's Validate method is called from any public API taking it as a param?

Reply:
Do we want the perf overhead of that on all service calls?  Would all models with properties added in later versions need to implement this?

Reply:
Most of the checks so far are happening in the LegacyServiceClient, so impact is minimal. Even still, this check is no more work that any argument checks we do.

Whatever and whenever we decide how we want to do this long-term, some maintainer can come back and make changes. I've isolated the brunt of the validation code (all the callees) to a separate "Validation" folder, though the namespace does not reflect that so no additional `using`s are needed in case we remove it.